mobile/src/main/java/org/openhab/habdroid/ui/preference/DeviceIdentifierPreference.kt

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.openhab.habdroid.ui.preference
+
+import android.content.Context
+import android.os.Build
+import android.text.Editable
+import android.text.InputType
+import android.text.TextWatcher
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.EditText
+import androidx.appcompat.app.AlertDialog
+import androidx.core.content.edit
+import androidx.core.os.bundleOf
+import androidx.fragment.app.DialogFragment
+import androidx.preference.DialogPreference
+import androidx.preference.PreferenceDialogFragmentCompat
+import com.google.android.material.switchmaterial.SwitchMaterial
+import com.google.android.material.textfield.TextInputLayout
+import org.openhab.habdroid.R
+import org.openhab.habdroid.util.PrefKeys
+
+class DeviceIdentifierPreference constructor(context: Context, attrs: AttributeSet) : DialogPreference(context, attrs) {
+    private var value: Any? = null
+
+    init {
+        dialogTitle = null
+        setPositiveButtonText(android.R.string.ok)
+        setNegativeButtonText(android.R.string.cancel)
+    }
+
+    override fun onSetInitialValue(defaultValue: Any?) {
+        value = defaultValue
+        updateSummary()
+    }
+
+    override fun getDialogLayoutResource(): Int {
+        return R.layout.pref_dialog_device_identifier
+    }
+
+    fun updateSummary() {
+        summary = if ((value as String?).isNullOrEmpty()) {
+            context.getString(R.string.device_identifier_summary_not_set)
+        } else {
+            value as String
+        }
+    }
+
+    fun setValue(value: String = (this.value as String?).orEmpty()) {
```

Comment:
Seeing this cast thrice, I wonder whether it would be worth adding a separate property (`valueString`) for the casted value.

Reply:
I prefer not to store the value twice. WDYT about https://github.com/openhab/openhab-android/pull/1983/commits/67661eb654a6c163b8f5400ba7176b1830423921?

Reply:
That one looks good to me.